### ISSUE TITLE
Weird occasional plugin failure mode - installed but not visible

### ISSUE THREAD

**Issue by noahgibbs** [jtimberman/vagrant-cookbook#23]
There's an odd failure case that has happened once to me and one of my users on Mac OS.

Specifically, a plugin can be installed (vagrant-host-shell for me, vagrant-omnibus for my user) so that the Vagrant cookbook thinks it's there, and later runs claim it's present and don't install it. But it doesn't show up on "vagrant plugin list" and it can't be used. When a Vagrantfile using it is run, and error occurs saying it can't use provider '' (empty string).

I'm still hunting this problem down. It seems to be rare and flaky rather than a consistent problem with a specific code snippet.

**Comment by noahgibbs** [comment]
I've tracked through a lot of this code now. Your code is just installing via "vagrant plugin install", so it may be a Vagrant bug, not a bug in your cookbook. But I'm definitely seeing a case where the plugin is installed, and it's in ~/.vagrant.d/plugins.json, but it's not showing properly on "vagrant plugin list". I have a cookbook ("https://github.com/noahgibbs/madscience_gem/blob/master/site-cookbooks/madscience-vagrant-cookbook/recipes/default.rb") that installs a number of Vagrant plugins... And by uninstalling Vagrant and running that cookbook repeatedly, I can get it wedged in this way repeatedly.

It's not every time, but it's not uncommon. At least when I install 5+ plugins at once.

The vagrant-host-shell plugin seems to be by far the most common culprit. Maybe something's weird about that plugin specifically somehow? It's also installed last, and one of the last ones I added to the cookbook, so hard to be sure.
